### PR TITLE
docs: How to fine-tune LLMs

### PR DESCRIPTION
**Pull request by hahuyhoang411**
## Describe Your Changes
Adding blogpost to docs/blog

- https://hackmd.io/kN8RuAGiS9SG5QbPyKYcyw

## Self Checklist

- [x] Transfer from notion over
- [x] Clarify some terms

### CODE REVIEW

**Comment by hahuyhoang411** [comment]
Personally, I think after creating open-foundry repo, I will create our own `GGUF converter` notebook instead of linking to other's.

**Comment by hahuyhoang411** [comment]
@0xSage I added you as a reviewer

cc @hieu-jan

File: docs/blog/finetune-with-docs.mdx [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+login(token=HUGGINGFACE_KEY_FILE)
+```
+
+## **2. Data generation**
+
+In this tutorial, the “data” we will use will be the [Nitro documentation](https://nitro.jan.ai/). Nitro is an open-source, lightweight (3 MB) inference server to supercharge apps with local AI.
+
+![About Nitro](img/about-nitro.png)
+
+**Fig 1. Example of Nitro’s documentation**
```
Comment (hieu-jan) [inline docs/blog/finetune-with-docs.mdx:78]: Typo in style, should we changed it to  

**Fig. 1: Example of Nitro's documentation.**

Comment (hahuyhoang411) [reply docs/blog/finetune-with-docs.mdx:78]: Good catch. It's a bug after I transfer the draft from notion

File: docs/blog/finetune-with-docs.mdx [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+To teach it, we will use a method called [**"Instruction tuning"**](#what-is-instruction-tuning), which involves refining the model's ability to comprehend and execute text-based instructions more effectively. 
+
+But how do we change the text from the documentation into the right data? We need to make pairs of questions and answers.
+
+The main idea is that we will break the documents into smaller pieces. Then, we use the LLM to help us develop questions and answers from those pieces. 
+
+*Note: We also open-source the [**Nitro’s documentation training dataset**](https://huggingface.co/datasets/jan-hq/nitro_binarized) in our [**Hugging Face Hub**](https://huggingface.co/jan-hq).*
+
+**Table 1. Example of Question and Answer pairs**
```
Comment (hieu-jan) [inline docs/blog/finetune-with-docs.mdx:90]: Should we move this line below the table 

**Table 1. Example of question and answer pairs.**

File: docs/blog/finetune-with-docs.mdx [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+```js title="Command to train LLM with alignment handbook"
+ACCELERATE_LOG_LEVEL=info \
+accelerate launch \
+  --config_file recipes/accelerate_configs/multi_gpu.yaml \
+  --num_processes=1 \
+  scripts/run_sft.py recipes/nitro/sft/config_lora.yaml
+```
+
+**Table 2. Training result of Nitro models**
```
Comment (hieu-jan) [inline docs/blog/finetune-with-docs.mdx:241]: Should we move this line below the table 

**Table 2. Training results of Nitro models.**

File: docs/blog/finetune-with-docs.mdx [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+*E: epochs, m: minutes*
+*Note: Training times can vary based on hardware specifications. The provided times are for reference purposes only.*
+
+## **4. Test the model**
+
+After training the model, it can be tested locally in the GGUF format using [Jan](https://jan.ai/). To convert the fine-tuned model to GGUF, you can utilize this convenient [Google Colab notebook by Maxime Labonne](https://colab.research.google.com/github/mlabonne/llm-course/blob/main/Quantize_Llama_2_models_using_GGUF_and_llama_cpp.ipynb).
+
+![Finetuned Nitro Respond](img/finetuned-response.png)
+
+**Fig 2. Using Jan to run a new fine-tuned model**
```
Comment (hieu-jan) [inline docs/blog/finetune-with-docs.mdx:259]: Typo in style, should we changed it to

**Fig. 2: Using Jan to run a new fine-tuned model.**

File: docs/blog/finetune-with-docs.mdx [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+![Finetuned Nitro Respond 2](img/finetuned-respond-2.png)
+
+**Fig 3. Model answers a technical question related to Nitro**
```
Comment (hieu-jan) [inline docs/blog/finetune-with-docs.mdx:263]: Typo in style, should we changed it to

**Fig. 3: Model answers a technical question related to Nitro.**

File: docs/blog/finetune-with-docs.mdx (hunk 2 above)
Comment (hahuyhoang411) [reply docs/blog/finetune-with-docs.mdx:90]: As I remembered correctly, the table's caption above the table is a common practice.

https://tex.stackexchange.com/questions/3243/why-should-a-table-caption-be-placed-above-the-table

File: docs/blog/finetune-with-docs.mdx [hunk 6]
```diff
@@ -0,0 +1,299 @@
+---
+title: "How to fine-tune LLMs - a tutorial"
+description: "Finetune LLMs with personal dataset"
+slug: /tutorial/how-to-fine-tune-llms
+tags: [Large Language Models, Finetune LLM, Personal dataset, Customzied dataset]
```
[the remark this answers is not in the document]
Comment (hieu-jan) [reply docs/blog/finetune-with-docs.mdx:5]: I extracted tags and SEO keywords based on current content and made a change.

**Review by hieu-jan — APPROVED** [review]
LGTM!

**Review by hieu-jan — APPROVED** [review]
Approved to move to pending blog post folders

**Pull request merged**